create_jenkins_job.py

```diff
@@ -249,7 +249,7 @@ def create_job(os_name, job_name, template_file, additional_dict):
                 'time_trigger_spec': PERIODIC_JOB_SPEC,
                 'mailer_recipients': DEFAULT_MAIL_RECIPIENTS + ' ros-contributions@amazon.com',
                 'build_args_default': data['build_args_default'] + ' --mixin tsan --packages-up-to test_communication',
-                'test_args_default': '--event-handlers console_direct+ --executor sequential --packages-select test_communication',
+                'test_args_default': '--event-handlers console_direct+ --executor sequential --packages-up-to rcutils',
```

Comment:
What motivates the change to testing packages-up-to rather than packages select? I don't think we need to run asan tests on rcutils dependencies, which are mostly the ament packages and launch as a test dependency.

Reply:
I see a few good reasons:

1. Keep ASAN and TSAN jobs aligned (reduce cognitive load)
1. Ensure that introducing a low-level package, it gets instrumented automatically. For instance, in your comment you forgot about `rcpputils` ;)

On a high-level note, my understanding is that we wanted to use `--packages-up-to` but reverted temporarily because doing that up to test communication was generating too much logs and made the JVM crash. Is my understanding wrong?

BTW I don't have a strong opinion for that, we can as well use a list of whitelisted packages if you prefer.

Reply:
I pushed an updated whitelisting test jobs one by one, this seems a better method indeed.